client/my-sites/site-settings/settings-security/controller.js

```diff
@@ -1,7 +1,17 @@
+import page from 'page';
 import { createElement } from 'react';
 import SecurityMain from 'calypso/my-sites/site-settings/settings-security/main';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 export function security( context, next ) {
-	context.primary = createElement( SecurityMain );
-	next();
+	const state = context.store.getState();
+	const siteId = getSelectedSiteId( state );
+	// If we have a site ID, render the main component
+	// Otherwise, redirect to site selection.
+	if ( siteId ) {
+		context.primary = createElement( SecurityMain );
+		next();
+	} else {
+		return page.redirect( '/settings/security' );
```

Comment:
Does this need to be returned? It works either way but returning seems safer.

Reply:
I don't know for sure -- I'd guess probably not, but this seems safe!

Reply:
Technically, we have to return just to prevent further execution, but there is nothing being executed afterward, so even if you don't return, it will be fine. As Noah mentioned, it's safer to return, because if anyone adds anything after the redirect, it will safely not be executed.

A minor detail here is that `page.redirect()` itself returns nothing, so I'd suggest moving `return` away from `page.redirect()`, in other words, moving the void `return` statement below the `page.redirect()` call, as done here:

https://github.com/Automattic/wp-calypso/blob/trunk/client/login/controller.js#L152 

However, since the value won't be used, it doesn't strongly matter - it's just more accurate because it properly separates the purpose of 2 separate actions (redirecting and aborting code execution) and that makes it more readable.

Reply:
Thanks, that makes sense! I moved the `return` to after `page.redirect()`.